### PR TITLE
Add context around Astronomer providers to deferrables instructions

### LINKED ISSUE #1905: Update deferrable operator implementation instructions to work for astronomer providers
**Affected Document:** https://docs.astronomer.io/learn/deferrable-operators

**Problem Type (Priority):**

- [ ] Error (Critical): This document has existing information that's incorrect and prevents readers from completing the documented setup (Add the `error/critical` label to this issue)
- [x] Misleading (Important): This document could benefit from additional context, guidance, or clarity (Add the `error/FYI` label to this issue)
- [ ] Typo (Low): Any issue related to spelling/ punctuation/ formatting (Add the `error/typo` label to this issue)

**Suggested Change (include links to relevant existing docs, support tickets, messages, or external content):** Update example(s) to include a note that if using Astronomer providers, a new package would need to be installed in `requirements.txt` and the package import path would change, which is beyond the currently stated instructions of simply swapping out the operator names.

**Additional Notes:** More context in this Slack thread: https://astronomer.slack.com/archives/C015V2JFKT5/p1676585879465719

### PR DESCRIPTION
**Pull request by kentdanas**
Resolves #1905

### CODE REVIEW

File: learn/deferrable-operators.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ With deferrable operators, worker slots are released when a task is polling for
...
 
-To use a deferrable version of an existing operator in your DAG, you only need to replace the import statement for the existing operator. For example, Airflow's `TimeSensorAsync` is a replacement of the non-deferrable `TimeSensor` ([source](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/sensors/time_sensor/index.html?highlight=timesensor#module-contents)). To use `TimeSensorAsync`, remove your existing `import` and replace it with the following:
+To use a deferrable version of an existing operator that is part of core Airflow in your DAG, you only need to replace the import statement for the existing operator. For example, Airflow's `TimeSensorAsync` is a replacement of the non-deferrable `TimeSensor` ([source](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/sensors/time_sensor/index.html?highlight=timesensor#module-contents)). To use `TimeSensorAsync`, remove your existing `import` and replace it with the following:
```
Comment (jwitz) [inline learn/deferrable-operators.md:40]:
```suggestion
To use a deferrable version of a core Airflow operator in your DAG, you only need to replace the import statement for the existing operator. For example, Airflow's `TimeSensorAsync` is a replacement of the non-deferrable `TimeSensor` ([source](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/sensors/time_sensor/index.html?highlight=timesensor#module-contents)). To use `TimeSensorAsync`, remove your existing `import` and replace it with the following:
```

File: learn/deferrable-operators.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ With deferrable operators, worker slots are released when a task is polling for
 
 Deferrable operators should be used whenever you have tasks that occupy a worker slot while polling for a condition in an external system. For example, using deferrable operators for sensor tasks can provide efficiency gains and reduce operational costs. Smart Sensors were deprecated in Airflow 2.2.4, and were removed in Airflow 2.4.0. Use deferrable operators instead of Smart Sensors because they provide greater functionality and they are supported by Airflow.
```
Comment (jwitz) [inline learn/deferrable-operators.md:38]: Now that we've had deferrable operators out for a while, I wanted to check in and ask: Is the strategy of `import <deferrable-operator> as <core-operator>` still best practices for all circumstances, or would this only be relevant in a more specific use case like migrating existing DAGs to be deferrable?

**Review by jwitz — APPROVED** [review]
I had one question I was curious about, but the core content LGTM!

File: learn/deferrable-operators.md (hunk 2 above)
Comment (kentdanas) [reply learn/deferrable-operators.md:38]: Good question - it's less a question of best practice and more that it makes it easier to update existing DAGs. While you definitely could do it this way in a new DAG, you wouldn't have to, and that's a personal preference. I added a note to try and clarify that - let me know what you think.

**Pull request merged**